metsim/methods/mtclim.py

```diff
@@ -157,42 +144,8 @@ def _calc_tfmax(prec, dtr, sm_dtr):
     # Calculate the diurnal temperature range
     df['t_max'] = np.maximum(df['t_max'], df['t_min'])
     dtr = df['t_max'] - df['t_min']
-    sm_dtr = pd.Series(dtr).rolling(
-        window=30, win_type=win_type, axis=0).mean().fillna(method='bfill')
-    if params['n_days'] <= 30:
-        warn('Timeseries is shorter than rolling mean window, filling ')
-        warn('missing values with unsmoothed data')
-        sm_dtr.fillna(dtr, inplace=True)
-
-    # Effective annual prec
-    if params['n_days'] <= 90:
-        # Simple scaled method, minimum of 8 cm
-        sum_prec = df['prec'].values.sum()
-        eff_ann_prec = (sum_prec / params['n_days']) * cnst.DAYS_PER_YEAR
-        eff_ann_prec = np.maximum(eff_ann_prec, 8.0)
-        parray = pd.Series(eff_ann_prec, index=df.index)
-    else:
-        # Calculate effective annual prec using 3 month moving window
-        window = pd.Series(np.zeros(params['n_days'] + 90))
-        window[90:] = df['prec']
-
-        # If yeardays at end match with those at beginning we can use
-        # the end of the input to generate the beginning by looping around
-        # If not, just duplicate the first 90 days
-        start_day, end_day = df.index.dayofyear[0], df.index.dayofyear[-1]
-        if ((start_day % 365 == (end_day % 365) + 1) or
-                (start_day % 366 == (end_day % 366) + 1)):
-            window[:90] = df['prec'][-90:]
-        else:
-            window[:90] = df['prec'][:90]
-
-        parray = cnst.DAYS_PER_YEAR * window.rolling(
-            window=90, win_type=win_type, axis=0).mean()[90:]
-
-    # Convert to cm
-    parray = parray.where(parray > 80.0, 80.0) / cnst.MM_PER_CM
-    # Doing this way because parray.reindex_like(df) returns all nan
-    parray.index = df.index
```

Comment:
This is now in `metsim.py` so that we can preprocess the whole dataset using the state file.